packages/roosterjs-content-model-plugins/lib/autoFormat/hyphen/transformHyphen.ts

```diff
@@ -13,12 +13,16 @@ export function transformHyphen(
     paragraph: ShallowMutableContentModelParagraph,
     context: FormatContentModelContext
 ): boolean {
+    // If the selection marker is at the beginning of a paragraph, there is no text to be transformed before it
+    if (paragraph.segments[0].segmentType == 'SelectionMarker') {
```

Comment:
I think instead of checking where is the selection marker, we should check if the event is already handled by some other feature (here it is switch to bullet list), an event should only be handled by one feature.

On the other hand, if we disabled the auto bullet feature, it is ok to transform hyphen here.